openedx/core/djangoapps/content_libraries/api/blocks.py

```diff
@@ -367,29 +384,36 @@ def create_library_block(
     return get_library_block(usage_key)
 
 
-def import_staged_content_from_user_clipboard(library_key: LibraryLocatorV2, user, block_id) -> XBlock:
+def _import_staged_block(
+    block_type: str,
+    olx_str: str,
+    library_key: LibraryLocatorV2,
+    source_context_key: LearningContextKey,
+    user,
+    staged_content_id: int,
+    staged_content_files: list[StagedContentFileData],
+    now: datetime,
+) -> LibraryXBlockMetadata:
     """
     Create a new library block and populate it with staged content from clipboard
 
     Returns the newly created library block
     """
     from openedx.core.djangoapps.content_staging import api as content_staging_api
-    if not content_staging_api:
-        raise RuntimeError("The required content_staging app is not installed")
 
-    user_clipboard = content_staging_api.get_user_clipboard(user)
-    if not user_clipboard:
-        return None
-
-    staged_content_id = user_clipboard.content.id
-    olx_str = content_staging_api.get_staged_content_olx(staged_content_id)
-    if olx_str is None:
-        return None  # Shouldn't happen since we checked that the clipboard exists - mostly here for type checker
-    staged_content_files = content_staging_api.get_staged_content_static_files(staged_content_id)
+    # Generate a block_id:
+    try:
+        olx_node = etree.fromstring(olx_str)
+        title = olx_node.attrib.get("display_name")
+        # Slugify the title and append some random numbers to make a unique slug
+        block_id = slugify(title, allow_unicode=True) + '-' + uuid4().hex[-6:]
```

Comment:
This is fine, but in the longer term, do you think it's worthwhile to add this concept to the Learning Core layer (find me some key that starts with this and doesn't collide)? At that layer, we could use things like the primary key to disambiguate them more succinctly.

Reply:
> This is fine, but in the longer term, do you think it's worthwhile to add this concept to the Learning Core layer (find me some key that starts with this and doesn't collide)?

Yes, that would be a nice improvement!